src/command/archive/get/get.c

```diff
@@ -200,74 +362,149 @@ archiveGetCheck(const StringList *archiveRequestList)
     ASSERT(archiveRequestList != NULL);
     ASSERT(!strLstEmpty(archiveRequestList));
 
-    ArchiveGetCheckResult result = {.archiveFileMapList = lstNewP(sizeof(ArchiveFileMap))};
+    ArchiveGetCheckResult result = {.archiveFileMapList = lstNewP(sizeof(ArchiveFileMap), .comparator = lstComparatorStr)};
 
     MEM_CONTEXT_TEMP_BEGIN()
     {
+        // List of warnings
+        StringList *warnList = strLstNew();
+
         // Get pg control info
         PgControl controlInfo = pgControlFromFile(storagePg());
 
-        // Get the repo storage in case it is remote and encryption settings need to be pulled down
-        storageRepo();
+        // Build list of repos/archiveIds where WAL may be found
+        List *cacheRepoList = lstNewP(sizeof(ArchiveGetFindCacheRepo));
 
-        result.cipherType = cipherType(cfgOptionStr(cfgOptRepoCipherType));
+        for (unsigned int repoIdx = 0; repoIdx < cfgOptionGroupIdxTotal(cfgOptGrpRepo); repoIdx++)
+        {
+            // If a repo was specified then skip all other repos
+            if (cfgOptionTest(cfgOptRepo) && cfgOptionUInt(cfgOptRepo) != cfgOptionGroupIdxToKey(cfgOptGrpRepo, repoIdx))
+                continue;
+
+            TRY_BEGIN()
+            {
+                // Get the repo storage in case it is remote and encryption settings need to be pulled down
+                storageRepoIdx(repoIdx);
 
-        // Attempt to load the archive info file
-        InfoArchive *info = infoArchiveLoadFile(
-            storageRepo(), INFO_ARCHIVE_PATH_FILE_STR, result.cipherType, cfgOptionStrNull(cfgOptRepoCipherPass));
+                ArchiveGetFindCacheRepo cacheRepo =
+                {
+                    .repoIdx = repoIdx,
+                    .cipherType = cipherType(cfgOptionIdxStr(cfgOptRepoCipherType, repoIdx)),
+                    .archiveList = lstNewP(sizeof(ArchiveGetFindCacheArchive)),
+                    .warnList = strLstNew(),
+                };
+
+                // Attempt to load the archive info file
+                InfoArchive *info = infoArchiveLoadFile(
+                    storageRepoIdx(repoIdx), INFO_ARCHIVE_PATH_FILE_STR, cacheRepo.cipherType,
+                    cfgOptionIdxStrNull(cfgOptRepoCipherPass, repoIdx));
+
+                MEM_CONTEXT_BEGIN(lstMemContext(result.archiveFileMapList))
+                {
+                    cacheRepo.cipherPassArchive = strDup(infoArchiveCipherPass(info));
```

Comment:
This will need to be added to each file in the list for that repo, so rather than dup it each time it makes sense to just assign it to the list context.

Reply:
So you're switching memory context here, duping the string and switching back, all to avoid strDup at lines 213 and 234?

Reply:
When you say it that way it does seem a bit excessive. I don't think there are any real performance implications here so we can just make those dups.

Reply:
As discussed, decided not to change this, but added a clarifying comment in d4a09984fb4981.